File: result/translate/transcoder/java2cpp/c++/COUNT_DISTINCT_NON_NEGATIVE_PAIRS_X_Y_SATISFY_INEQUALITY_XX_YY_N_2_1.cpp
int f_gold ( int n ) {
  int x = 0 , y_count , res = 0 ;
  for ( y_count = 0 ;
  y_count * y_count < n ;
  y_count ++ ) ;
  while ( y_count != 0 ) {
    res += y_count ;
    x ++ ;
    while ( y_count != 0 && ( x * x + ( y_count - 1 ) * ( y_count - 1 ) >= n ) ) {
      y_count -- ;
    }
  }
  return res ;
}

File: result/translate/transcoder/java2python/python/COUNT_DISTINCT_NON_NEGATIVE_PAIRS_X_Y_SATISFY_INEQUALITY_XX_YY_N_2_1.py
def f_gold ( n ) :
    x , y_count , res = 0 , 0 , 0
    for y_count in range ( 0 , y_count * y_count < n ) :
        pass
    while y_count != 0 :
        res += y_count
        x += 1
        while y_count != 0 and ( x * x + ( y_count - 1 ) * ( y_count - 1 ) >= n ) :
            y_count -= 1
    return res
